modules/internal/ChapelBase.chpl

```diff
@@ -1286,23 +1306,29 @@ module ChapelBase {
     __primitive("/=", lhs, rhs);
   }
   inline proc /=(ref lhs:real(?w), rhs:real(w)) {
+    if (chpl_checkDivByZero) then
+      if rhs == 0.0 then
+        halt("Attempt to divide by zero");
     __primitive("/=", lhs, rhs);
   }
   inline proc /=(ref lhs, rhs) {
     lhs = lhs / rhs;
   }
 
   inline proc %=(ref lhs:int(?w), rhs:int(w)) {
+    if (chpl_checkDivByZero) then
+      if rhs == 0:int(w) then
+        halt("Attempt to divide by zero");
     __primitive("%=", lhs, rhs);
   }
   inline proc %=(ref lhs:uint(?w), rhs:uint(w)) {
-    __primitive("%=", lhs, rhs);
-  }
-  inline proc %=(ref lhs:real(?w), rhs:real(w)) {
+    if (chpl_checkDivByZero) then
+      if rhs == 0:uint(w) then
+        halt("Attempt to divide by zero");
     __primitive("%=", lhs, rhs);
   }
   inline proc %=(ref lhs, rhs) {
-    lhs = lhs % rhs;
+      lhs = lhs % rhs;
```

Comment:
Unless I'm misreading this diff, it looks to me like this is unusually indented in the new version.